### PR TITLE
Make it a proper python package

### PR DESCRIPTION
**Pull request by rnsc**
Hello,

I wanted to have a proper Python package for this for an easier install process with Ansible.
Obviously the URL in install.sh file would need to change for you repository.

It can be built with a simple

```shell
python3 -m build
```

A GitHub workflow could even be setup to create releases automatically.

### CODE REVIEW

File: install.sh [hunk 1]
```diff
@@ -2,15 +2,14 @@
...
 
-echo "settling to /usr/local/bin/automagic-fan/..."
-rm -r /usr/bin/automagic-fan/ 2>/dev/null
-mkdir -p /usr/local/bin/automagic-fan
-cp fanctl.py /usr/local/bin/automagic-fan/
-echo "done"
+apt-get update
+apt-get install python3-pip
+
+pip3 install "https://github.com/rnsc/jetson-fan-ctl/releases/download/v0.1.0/jetsonfanctl-0.1.0-py3-none-any.whl"
```
Comment (rnsc) [inline install.sh:12]: This URL would need to be changed.

**Comment by Pyrestone** [comment]
It's certainly an interesting idea.

I think it would be neat to offer a one-line 'pip install jetson-fan-ctl' from pip/pypi.

I'll have to look at it over the weekend.

Currently, we don't have pip as a dependecy, so this might break some people's installation.

Maybe we can keep the original way of installation as an alternative for people who don't have pip for whatever reason (i think it doesn't come stock on the nano) or no internet on the machine.

**Comment by rnsc** [comment]
I'm open to make any changes to this PR or we can discard it.
I did a release of a wheel package in my fork so that I could already automate it for my project, but I figured I could contribute my small piece of work.

To make it confirm to PyPi, some stuff need to be filled in first, but there shouldn't be too much work on top of this.

Regarding the local instance, we could mimick what I did with the package then and have the Python file named as a standard binary instead of with the .py extension?

**Comment by rnsc** [comment]
> I think it would be neat to offer a one-line 'pip install jetson-fan-ctl' from pip/pypi.
> 

Side note, dashes in package name are forbidden, thus why I renamed it. Wondering if we provide a pip package if we shouldn't rename the systemd unit file too to match that name?

**Comment by rnsc** [comment]
> 
> > I think it would be neat to offer a one-line 'pip install jetson-fan-ctl' from pip/pypi.
> > 
> 
> Side note, dashes in package name are forbidden, thus why I renamed it. Wondering if we provide a pip package if we shouldn't rename the systemd unit file too to match that name?

Strike that, I'm wrong apparently there pypi packages with dashes in their names...

**Comment by rnsc** [comment]
Hello @Pyrestone, wondering if you had time to take a look at making this happen? Should I just my PR? Thank you

**Comment by rnsc** [comment]
Closing this to cleanup my PR list

**Pull request closed without merging**